mono/utils/mono-mmap.c

```diff
@@ -81,18 +81,36 @@ aligned_address (char *mem, size_t size, size_t alignment)
 	return aligned;
 }
 
-static volatile size_t allocation_count [MONO_MEM_ACCOUNT_MAX];
+static size_t allocation_count [MONO_MEM_ACCOUNT_MAX];
+static size_t total_allocation_count;
+static size_t alloc_limit;
 
 void
 account_mem (MonoMemAccountType type, ssize_t size)
 {
 #if SIZEOF_VOID_P == 4
 	InterlockedAdd ((volatile gint32*)&allocation_count [type], (gint32)size);
+	InterlockedAdd ((volatile gint32*)&total_allocation_count, (gint32)size);
```

Comment:
What about adding a size_t overload to InterlockedAdd?

It would require us breaking out of the win32 API for atomics, but I think this is a long time wish of @alexrp.